src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxQueryEvaluator.cs

```diff
@@ -305,6 +320,151 @@ private string GetField(FieldExpression f)
+                    var orderTypeField = OrderTypeFieldConverter(orders[0].FieldType);
+
+                    match = orders[0].Ascending
+                        ? _searcher.OrderByAscending(match, id, orderTypeField, take)
+                        : _searcher.OrderByDescending(match, id, orderTypeField, take);
+
+
+                    return match;
+                }
+                case 2:
+                {
+                    var firstOrder = orders[0];
+                    var secondOrder = orders[1];
+                    if (firstOrder.Expression is not FieldExpression first || secondOrder.Expression is not FieldExpression second)
+                    {
+                        throw new Exception();
+                    }
+                    var firstId = GetFieldIdInIndex(GetField(first));
+                    var firstTypeField = OrderTypeFieldConverter(firstOrder.FieldType);
+                    var secondId = GetFieldIdInIndex(GetField(second));
+                    var secondTypeField = OrderTypeFieldConverter(secondOrder.FieldType);
+
+                    return (firstOrder.Ascending, secondOrder.Ascending) switch
+                    {
+                        (true, true) => SortingMultiMatch.Create(_searcher, match, 
+                            new SortingMatch.AscendingMatchComparer(_searcher, firstId, firstTypeField),
+                            new SortingMatch.AscendingMatchComparer(_searcher, secondId, secondTypeField)),
+                    
+                        (false, true) => SortingMultiMatch.Create(_searcher, match, 
+                            new SortingMatch.DescendingMatchComparer(_searcher, firstId, firstTypeField),
+                            new SortingMatch.AscendingMatchComparer(_searcher, secondId, secondTypeField)),
+                    
+                        (true, false) => SortingMultiMatch.Create(_searcher, match, 
+                            new SortingMatch.AscendingMatchComparer(_searcher, firstId, firstTypeField),
+                            new SortingMatch.DescendingMatchComparer(_searcher, secondId, secondTypeField)),
+                    
+                        (false, false) => SortingMultiMatch.Create(_searcher, match, 
+                            new SortingMatch.DescendingMatchComparer(_searcher, firstId, firstTypeField),
+                            new SortingMatch.DescendingMatchComparer(_searcher, secondId, secondTypeField))
+                    };
+                }
+                case 3:
+                {
+                    var firstOrder = orders[0];
+                    var secondOrder = orders[1];
+                    var thirdOrder = orders[2];
+
+                    if (firstOrder.Expression is not FieldExpression first || secondOrder.Expression is not FieldExpression second ||
+                        thirdOrder.Expression is not FieldExpression third)
+                    {
+                        throw new Exception();
```

Comment:
And here - better message

Reply:
https://github.com/ravendb/ravendb/blob/b155b9ddcf8a5dd4ae877a565d3ce0390997a6e6/src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxQueryEvaluator.cs#L303

Something like this, for now. I'm pretty sure this will be changed in the future anyway.